tests/dialects/test_op_builder.py

```diff
@@ -77,3 +79,95 @@ def region(b: Builder, args: tuple[BlockArgument, ...]):
     assert isinstance(ops[1], Constant)
     assert isinstance(ops[1].value, IntegerAttr)
     assert ops[1].value.value is two
+
+
+def test_build_implicit_region():
+
+    one = IntAttr(1)
+    two = IntAttr(2)
+
+    @Builder.implicit_region
+    def region():
+        Constant.from_int_and_width(one, i32)
+        Constant.from_int_and_width(two, i32)
+
+    assert len(region.blocks) == 1
+
+    ops = region.ops
+
+    assert len(ops) == 2
+
+    assert isinstance(ops[0], Constant)
+    assert isinstance(ops[0].value, IntegerAttr)
+    assert ops[0].value.value is one
+
+    assert isinstance(ops[1], Constant)
+    assert isinstance(ops[1].value, IntegerAttr)
+    assert ops[1].value.value is two
+
+
+def test_build_implicit_callable_region():
+
+    one = IntAttr(1)
+    two = IntAttr(2)
+
+    @Builder.implicit_region([i32])
+    def region(args: tuple[BlockArgument, ...]):
+        assert len(args) == 1
```

Comment:
For example here, I would find something like this more readable:
```
@Builder.implicit_region([i32])
def region(arg: BlockArgument):
   ...
```
Bonus point if we can type that, but I guess this would require us to only pass tuples to `Builder.implicit_region`, so probably won't work.
I think we should still be able to inline `args` in a typing fashion, since we don't have any other arguments.